strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -28,13 +28,27 @@
 
 from strawberryfields.backends.bosonicbackend.bosoniccircuit import BosonicModes
 from strawberryfields.backends.base import NotApplicableError
+from strawberryfields.program_utils import CircuitError
+import sympy
 
 
 def kron_list(l):
     """Take Kronecker products of a list of lists."""
     return reduce(np.kron, l)
 
 
+def parameter_checker(parameters):
+    """Checks if any items in an iterable are sympy objects."""
+    for item in parameters:
+        if isinstance(item, sympy.Expr):
+            return True
+        # This checks all the items within item if item is a list, array, etc.
+        if hasattr(item, "__iter__") and not isinstance(item, str):
+            if parameter_checker(item):
+                return True
+    return False
```

Comment:
I'm a bit surprised you have to have this check - were there cases where sympy objects were propagating down to the backend?

Reply:
The MeasuredParameter and FreeParameter classes -- and if any operations are applied to those objects -- are sympy expressions. 
This came up when I ran test_parameters_integration in the integration tests, wherein the phase of creating a cat state is conditioned on the value of a measurement. 
For the moment, we are not considering conditional non-Gaussian state prep like that (since we initialize all the non-Gaussian states at the beginning of the circuit), so I have this function to check if any of the parameters used to initialize those states are symbolic objects.

Reply:
got it!